pyomo/core/base/reference.py

```diff
@@ -348,62 +387,103 @@ def _get_iter(self, _slice, key):
 
 
 def _identify_wildcard_sets(iter_stack, index):
-    # if we have already decided that there isn't a comon index for the
+    # if we have already decided that there isn't a common index for the
     # slices, there is nothing more we can do.  Bail.
     if index is None:
         return index
 
     # Walk the iter_stack that led to the current item and try to
     # identify the component wildcard sets
+    #
+    # `tmp` will be a list of dicts mirroring the iter_stack. Each
+    # dict maps position within that level's component's "subsets"
+    # list to the set at that position if it is a wildcard set.
     tmp = [None]*len(iter_stack)
     for i, level in enumerate(iter_stack):
         if level is not None:
             offset = 0
+            # `offset` is the position, within the total index tuple,
+            # of the first coordinate of a set.
+            # This function seems to only make sense if 
+            # normalize_index.flatten is True.
```

Comment:
This comment should probably be promoted to to the method documentation and not buried down here.